core/src/test/java/com/graphhopper/routing/util/BikeFlagEncoderTest.java

```diff
@@ -68,23 +68,40 @@ public void testGetSpeed()
         way.clearTags();
         way.setTag("highway", "footway");
         way.setTag("bicycle", "yes");
-        assertEquals(6, encoder.getSpeed(way));
+        assertEquals(PUSHING_SECTION_SPEED * 2, encoder.getSpeed(way));
         assertPriority(UNCHANGED.getValue(), way);
 
         way.clearTags();
         way.setTag("highway", "footway");
         way.setTag("surface", "paved");
         way.setTag("bicycle", "yes");
 
-        assertEquals(6, encoder.getSpeed(way));
+        assertEquals(PUSHING_SECTION_SPEED * 2, encoder.getSpeed(way));
         assertPriority(UNCHANGED.getValue(), way);
 
         way.clearTags();
+        way.setTag("highway", "cycleway");
+        assertEquals(18, encoder.getSpeed(way));
+        assertPriority(VERY_NICE.getValue(), way);
+        int cyclewayspeed = encoder.getSpeed(way);
```

Comment:
Please check the whole code and not only the diff. I suppose you are missing an assert after the assignment to `cyclewayspeed`, but I'm just fetching it and use it in further tests below.